src/components/tikzpictureview.ts

```diff
@@ -0,0 +1,361 @@
+        }
+
+        this.checkPreamble(tikzFileCollection)
+
+        const tikzPictures: IFileTikzPicture[] = tikzFileCollection.tikzPictures
+
+        for (const tikzPicture of tikzPictures) {
+            if (
+                this.extension.viewer.clients[tikzPicture.tempFile.replace(/\.tex$/, '.pdf').toLocaleUpperCase()]
+                    .length === 0
+            ) {
+                tikzPictures.splice(tikzPictures.indexOf(tikzPicture), 1)
+                this.cleanupTikzPicture(tikzPicture)
+                continue
+            }
+
+            const tikzPicturesToUpdate: IFileTikzPicture[] = []
+
+            for (const change of changes) {
+                const lineDelta =
+                    change.text.split('\n').length - 1 - (change.range.end.line - change.range.start.line)
+                if (tikzPicture.range.end.isBefore(change.range.start)) {
+                    // change after tikzpicture
+                    continue
+                } else if (tikzPicture.range.start.isAfter(change.range.end)) {
+                    // change before tikzpicture
+                    tikzPicture.range = new vscode.Range(
+                        tikzPicture.range.start.translate(lineDelta, 0),
+                        tikzPicture.range.end.translate(lineDelta, 0)
+                    )
+                } else if (
+                    change.range.end.isAfter(tikzPicture.range.start) &&
+                    change.range.end.line - lineDelta < tikzPicture.range.start.line
+                ) {
+                    // tikzpicture removed
+                    tikzPictures.splice(tikzPictures.indexOf(tikzPicture), 1)
+                    this.cleanupTikzPicture(tikzPicture)
+                } else {
+                    // tikzpicture modified
+                    let startLocation: vscode.Position | null = null
+                    if (change.range.start.line <= tikzPicture.range.start.line) {
+                        const startLine = document.lineAt(tikzPicture.range.start.line)
+                        const tikzPictureStartIndex = startLine.text.indexOf('\\begin{tikzpicture}')
+                        if (tikzPictureStartIndex !== -1) {
+                            startLocation = tikzPicture.range.start.translate(
+                                0,
+                                tikzPictureStartIndex - tikzPicture.range.start.character
+                            )
+                        } else {
+                            const startRegex = /\\begin{tikzpicture}/
+                            let startMatch: RegExpMatchArray | null = null
+                            let lineNo = change.range.start.line - 1
+                            do {
+                                startMatch = document.lineAt(++lineNo).text.match(startRegex)
+                            } while (!startMatch && lineNo <= tikzPicture.range.end.line)
+
+                            if (startMatch) {
+                                // @ts-ignore
```

Comment:
Don't use @ts-ignore. Fix it.

Reply:
See 86daf0ec53d032b66a5fa7a69bfbd9845ab2cdc2